### PR TITLE
Transactional shorthand until we get the 2.12 Java Lambda interop

### PR DESCRIPTION
**Pull request by joan38**
graph.transactional { graph =>
      graph.addVertex(...)
      graph.addVertex(...)
      ...
    }

does all actions in ACID

### CODE REVIEW

File: gremlin-scala/src/main/scala/gremlin/scala/ScalaGraph.scala [hunk 1]
```diff
@@ -89,4 +90,8 @@ case class ScalaGraph[G <: Graph](graph: G) {
   def compute(): GraphComputer = graph.compute()
 
   def close(): Unit = graph.close()
+
+  def transactional[R](work: Graph => R) = graph.tx.submit(new JFunction[Graph, R] {
+    def apply(g: Graph) = work(g)
```
Comment (mpollmeier) [inline gremlin-scala/src/main/scala/gremlin/scala/ScalaGraph.scala]: you could also use `toJavaFunction` from the package object

**Comment by mpollmeier** [comment]
LGTM

File: gremlin-scala/src/main/scala/gremlin/scala/ScalaGraph.scala (hunk 1 above)
Comment (joan38) [inline gremlin-scala/src/main/scala/gremlin/scala/ScalaGraph.scala]: Done

**Comment by joan38** [comment]
That's useful for now but I think once we have 2.12 and ditched 2.11 (pfiou) that method will not have anything better than `.tx.submit()`

**Comment by mpollmeier** [comment]
agreed. by then we can drop a number of helper methods from the package object.

**Pull request merged**